Content.Server/Speech/Components/ListenWireAction.cs

```diff
@@ -0,0 +1,116 @@
+using System.Text;
+
+using Content.Server.Speech.Components;
+using Content.Server.Speech.EntitySystems;
+using Content.Server.VoiceMask;
+using Content.Server.Wires;
+using Content.Shared.Speech;
+using Content.Shared.Wires;
+
+namespace Content.Server.Speech;
+
+public sealed partial class ListenWireAction : BaseToggleWireAction
+{
+    private WiresSystem _wires = default!;
+
+    /// <summary>
+    /// Length of the gibberish string sent when pulsing the wire
+    /// </summary>
+    private int _noiseLength = 16;
+    public override Color Color { get; set; } = Color.Green;
+    public override string Name { get; set; } = "wire-name-listen";
+
+    public override object? StatusKey { get; } = ListenWireActionKey.StatusKey;
+
+    public override object? TimeoutKey { get; } = ListenWireActionKey.TimeoutKey;
+
+    public override int Delay { get; } = 10;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _wires = EntityManager.System<WiresSystem>();
+    }
+    public override StatusLightState? GetLightState(Wire wire)
+    {
+        if (GetValue(wire.Owner))
+            return StatusLightState.On;
+        else
+        {
+            if (TimeoutKey != null && _wires.HasData(wire.Owner, TimeoutKey))
+                return StatusLightState.BlinkingSlow;
+            return StatusLightState.Off;
+        }
+    }
+    public override void ToggleValue(EntityUid owner, bool setting)
+    {
+        if (setting)
+        {
+            // If we defer removal, the status light gets out of sync
+            EntityManager.RemoveComponent<BlockListeningComponent>(owner);
+        }
+        else
+        {
+            EntityManager.EnsureComponent<BlockListeningComponent>(owner, out _);
+        }
+    }
+
+    public override bool GetValue(EntityUid owner)
+    {
+        return !EntityManager.HasComponent<BlockListeningComponent>(owner);
+    }
+
+    public override void Pulse(EntityUid user, Wire wire)
+    {
+        if (!GetValue(wire.Owner) || !IsPowered(wire.Owner))
+            return;
+
+        base.Pulse(user, wire);
+
+        // We have to use a valid euid in the ListenEvent. The user seems
+        // like a sensible choice, but we need to mask their name.
+
+        // Save the user's existing voicemask if they have one
+        bool oldEnabled = true;
+        string oldVoiceName = "ERROR";
+        if (EntityManager.TryGetComponent<VoiceMaskComponent>(user, out var oldMask))
```

Comment:
This needs to be localised.

Reply:
Yep, done. In theory it should never be shown to the player, but it makes sense to play it safe.